src/Sulu/Bundle/ActivityBundle/UserInterface/Controller/ActivityController.php

```diff
@@ -0,0 +1,472 @@
+{
+    use RequestParametersTrait;
+
+    /**
+     * @var DoctrineListBuilderFactoryInterface
+     */
+    private $listBuilderFactory;
+
+    /**
+     * @var FieldDescriptorFactoryInterface
+     */
+    private $fieldDescriptorFactory;
+
+    /**
+     * @var RestHelperInterface
+     */
+    private $restHelper;
+
+    /**
+     * @var SecurityCheckerInterface
+     */
+    private $securityChecker;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string
+     */
+    private $activityClass;
+
+    /**
+     * @var string
+     */
+    private $contactClass;
+
+    /**
+     * @var string
+     */
+    private $userClass;
+
+    /**
+     * @var array<string, int>
+     */
+    private $permissions;
+
+    /**
+     * @param array<string, int> $permissions Inject `sulu_security.permissions` parameter
+     */
+    public function __construct(
+        FieldDescriptorFactoryInterface $fieldDescriptorFactory,
+        DoctrineListBuilderFactoryInterface $listBuilderFactory,
+        RestHelperInterface $restHelper,
+        SecurityCheckerInterface $securityChecker,
+        TranslatorInterface $translator,
+        string $activityClass,
+        string $contactClass,
+        string $userClass,
+        array $permissions,
+        ViewHandlerInterface $viewHandler,
+        ?TokenStorageInterface $tokenStorage = null
+    ) {
+        parent::__construct($viewHandler, $tokenStorage);
+
+        $this->fieldDescriptorFactory = $fieldDescriptorFactory;
+        $this->listBuilderFactory = $listBuilderFactory;
+        $this->restHelper = $restHelper;
+        $this->securityChecker = $securityChecker;
+        $this->translator = $translator;
+        $this->activityClass = $activityClass;
+        $this->contactClass = $contactClass;
+        $this->userClass = $userClass;
+        $this->permissions = $permissions;
+    }
+
+    public function cgetAction(Request $request): Response
+    {
+        /** @var UserInterface $user */
+        $user = $this->getUser();
+
+        /** @var array<string, FieldDescriptorInterface> $configurationFieldDescriptors */
+        $configurationFieldDescriptors = $this->fieldDescriptorFactory->getFieldDescriptors('activities');
+
+        $fieldDescriptors = \array_merge(
+            $this->getRawDataFieldDescriptors(),
+            $configurationFieldDescriptors
+        );
+
+        /** @var DoctrineListBuilder $listBuilder */
+        $listBuilder = $this->listBuilderFactory->create($this->activityClass);
+        $this->restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
+        $listBuilder->setSelectFields($fieldDescriptors);
+        $listBuilder->sort($fieldDescriptors['timestamp'], ListBuilderInterface::SORTORDER_DESC);
+
+        $translationLocale = $user->getLocale();
+
+        /** @var string|null $resourceLocale */
+        $resourceLocale = $this->getRequestParameter($request, 'locale');
```

Comment:
in other APIs, the `locale` parameter will change the language of the data is returned, in this it will act similar to a filter. maybe we should call it `resourceLocale` because of the different behaviour?

Reply:
that would allow us to add a `locale` parameter to change the locale that is used by the translator if we need to do that in the future

Reply:
Yeah i think its good that it matches the locale of the user 🙂  What I wanted to say is that I think I would expect the `locale` parameter to affect the language of the text like in other APIs, not filter the items.

I think its possible to configure the list view to send the locale of the route (eg. in the page activities) via another parameter with something like `->addRouterAttributesToListRequest(['llocale' => 'resourceLocale'])`

Reply:
Unfortunately, we figured that `addRouterAttributesToListRequest` does not work if the locale is switched after the initial rendering of the list. Therefore we are going to use `locale` 🙂